All connections using Spectra S3 connection profiles fail after upgrade from 9.2.4 to 9.3.x

**Describe the bug**
Existing connection bookmarks to Spectra S3  (BlackPearl, not standard AWS S3) fail to list buckets post 9.3.0 upgrade.
This appears to be due to a failure in SpectraListService.:

```
2026-02-16 12:36:41,576 [background-5] DEBUG ch.cyberduck.core.worker.ListWorker - Run with feature VaultRegistryListService{proxy=ch.cyberduck.core.spectra.SpectraListService@17f193e}
2026-02-16 12:36:41,576 [background-5] DEBUG ch.cyberduck.core.vault.registry.VaultRegistryListService - Look for vault in Path{path='/', type=[directory, volume]}

```
```
2026-02-16 12:36:41,592 [background-5] INFO  com.spectralogic.ds3client.networking.NetworkClientImpl - Sending request: GET http://10.11.12.31:80 /
2026-02-16 12:36:41,616 [background-5] WARN  ch.cyberduck.core.threading.BackgroundCallable - Failure running background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
java.lang.Exception
```

**To Reproduce**
Upgrade a 9.2.4 install with working bookmarks to 9.3.0 or 9.3.1


**Expected behavior**
Bucket listing on connection

**Screenshots**
No log window or bucket list appears after connection

**Desktop (please complete the following information):**
 - OS: Windows 111

**Log Files**

Working connection at 9.2.4:
```

2026-02-16 12:35:21,268 [background-5] INFO  ch.cyberduck.transcript.request - Host: 10.11.12.31
2026-02-16 12:35:21,272 [background-5] INFO  ch.cyberduck.transcript.request - Connection: Keep-Alive
2026-02-16 12:35:21,272 [background-5] INFO  ch.cyberduck.transcript.request - User-Agent: Cyberduck/9.2.4.43667 (Windows 11/10.0.22631.0) (amd64)
2026-02-16 12:35:21,272 [background-5] INFO  ch.cyberduck.transcript.request - Accept-Encoding: gzip,deflate
2026-02-16 12:35:21,272 [background-5] INFO  ch.cyberduck.transcript.request - x-amz-date: 20260216T193521Z
2026-02-16 12:35:21,272 [background-5] INFO  ch.cyberduck.transcript.request - Authorization: ********
2026-02-16 12:35:21,304 [background-5] INFO  ch.cyberduck.transcript.response - HTTP/1.1 200 OK
2026-02-16 12:35:21,308 [background-5] INFO  ch.cyberduck.transcript.response - RequestHandler-Version: 1.8D18E6B210A254D12ACD074CF265700F
2026-02-16 12:35:21,308 [background-5] INFO  ch.cyberduck.transcript.response - x-amz-request-id: 3049
2026-02-16 12:35:21,308 [background-5] INFO  ch.cyberduck.transcript.response - Content-Type: text/xml;charset=UTF-8
2026-02-16 12:35:21,309 [background-5] INFO  ch.cyberduck.transcript.response - Content-Language: en-US
2026-02-16 12:35:21,309 [background-5] INFO  ch.cyberduck.transcript.response - Content-Length: 666
2026-02-16 12:35:21,309 [background-5] INFO  ch.cyberduck.transcript.response - Date: Mon, 16 Feb 2026 19:35:21 GMT
2026-02-16 12:35:21,309 [background-5] INFO  ch.cyberduck.transcript.response - Keep-Alive: timeout=60
2026-02-16 12:35:21,309 [background-5] INFO  ch.cyberduck.transcript.response - Connection: keep-alive
2026-02-16 12:35:21,320 [background-5] DEBUG org.apache.http.impl.execchain.MainClientExec - Connection can be kept alive for 60000 MILLISECONDS
2026-02-16 12:35:21,324 [background-5] DEBUG org.jets3t.service.impl.rest.httpclient.RestStorageService - Response for 'GET'. Content-Type: text/xml;charset=UTF-8, Headers: [RequestHandler-Version: 1.8D18E6B210A254D12ACD074CF265700F, x-amz-request-id: 3049, Content-Type: text/xml;charset=UTF-8, Content-Language: en-US, Content-Length: 666, Date: Mon, 16 Feb 2026 19:35:21 GMT, Keep-Alive: timeout=60, Connection: keep-alive]
2026-02-16 12:35:21,324 [background-5] DEBUG org.jets3t.service.impl.rest.httpclient.RestStorageService - Response entity: ResponseEntityProxy{[Content-Type: text/xml;charset=UTF-8,Content-Length: 666,Chunked: false]}
2026-02-16 12:35:21,324 [background-5] DEBUG org.jets3t.service.impl.rest.httpclient.RestStorageService - Entity length: 666
2026-02-16 12:35:21,324 [background-5] DEBUG org.jets3t.service.impl.rest.httpclient.RestStorageService - Received response code 200; matches one of expected set ([200])? true
2026-02-16 12:35:21,384 [background-5] DEBUG org.jets3t.service.Jets3tProperties - xmlparser.sanitize-listings=false
2026-02-16 12:35:21,384 [background-5] DEBUG org.jets3t.service.impl.rest.XmlResponsesSaxParser - Parsing XML response document with handler: class org.jets3t.service.impl.rest.XmlResponsesSaxParser$ListAllMyBucketsHandler
2026-02-16 12:35:21,492 [background-5] DEBUG org.jets3t.service.impl.rest.XmlResponsesSaxParser - Created new bucket from listing: S3Bucket [name=Bobdest,location=null,creationDate=Thu Sep 11 11:56:47 MDT 2025,owner=null] Metadata={Date=Thu Sep 11 11:56:47 MDT 2025}
2026-02-16 12:35:21,492 [background-5] DEBUG org.jets3t.service.impl.rest.XmlResponsesSaxParser - Created new bucket from listing: S3Bucket [name=Bobsource,location=null,creationDate=Thu Sep 11 11:51:50 MDT 2025,owner=null] Metadata={Date=Thu Sep 11 11:51:50 MDT 2025}
2026-02-16 12:35:21,492 [background-5] DEBUG org.jets3t.service.impl.rest.XmlResponsesSaxParser - Created new bucket from listing: S3Bucket [name=Jim,location=null,creationDate=Mon Sep 15 11:40:51 MDT 2025,owner=null] Metadata={Date=Mon Sep 15 11:40:51 MDT 2025}
2026-02-16 12:35:21,492 [background-5] DEBUG org.jets3t.service.impl.rest.XmlResponsesSaxParser - Created new bucket from listing: S3Bucket [name=Spectra-BlackPearl-Backup-slab-boneyard-500304800913b93f,location=null,creationDate=Mon Jun 30 09:07:48 MDT 2025,owner=null] Metadata={Date=Mon Jun 30 09:07:48 MDT 2025}
2026-02-16 12:35:21,492 [background-5] DEBUG org.jets3t.service.impl.rest.XmlResponsesSaxParser - Created new bucket from listing: S3Bucket [name=testbucketjet,location=null,creationDate=Mon Sep 22 15:00:30 MDT 2025,owner=null] Metadata={Date=Mon Sep 22 15:00:30 MDT 2025}
2026-02-16 12:35:21,496 [background-5] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection [id: 0][route: {}->http://10.11.12.31:80] can be kept alive for 60.0 seconds
2026-02-16 12:35:21,496 [background-5] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-0: set socket timeout to 0
2026-02-16 12:35:21,500 [background-5] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection released: [id: 0][route: {}->http://10.11.12.31:80][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
2026-02-16 12:35:21,500 [background-5] DEBUG org.jets3t.service.impl.rest.httpclient.HttpMethodReleaseInputStream - Released HttpMethod as its response data stream is fully consumed
2026-02-16 12:35:21,504 [background-5] DEBUG ch.cyberduck.core.KeychainLoginService - Login successful for Session{host=Host{protocol=Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}, region='null', port=80, hostname='10.11.12.31', credentials=Credentials{user='bmV3YWRtaW4=', password='********', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=9223372036854775807}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=9223372036854775807}', identity=null}, uuid='7b39f090-42dd-40be-b0bd-129d1af6e7eb', nickname='null', defaultpath='null', workdir=null, custom=null, labels=null}, state=open}
2026-02-16 12:35:21,506 [background-5] INFO  ch.cyberduck.core.KeychainLoginService - Skip writing credentials for bookmark 10.11.12.31
2026-02-16 12:35:21,506 [background-5] DEBUG ch.cyberduck.core.KeychainLoginService - Reset credentials for Host{protocol=Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}, region='null', port=80, hostname='10.11.12.31', credentials=Credentials{user='bmV3YWRtaW4=', password='********', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=9223372036854775807}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=9223372036854775807}', identity=null}, uuid='7b39f090-42dd-40be-b0bd-129d1af6e7eb', nickname='null', defaultpath='null', workdir=null, custom=null, labels=null}
2026-02-16 12:35:21,506 [background-5] DEBUG ch.cyberduck.core.threading.WorkerBackgroundAction - Run worker MountWorker{cache=Cache{size=0}}
2026-02-16 12:35:21,513 [background-5] DEBUG ch.cyberduck.core.shared.WorkdirHomeFeature - No workdir set for bookmark Host{protocol=Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}, region='null', port=80, hostname='10.11.12.31', credentials=Credentials{user='bmV3YWRtaW4=', password='', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=9223372036854775807}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=9223372036854775807}', identity=null}, uuid='7b39f090-42dd-40be-b0bd-129d1af6e7eb', nickname='null', defaultpath='null', workdir=null, custom=null, labels=null}
2026-02-16 12:35:21,513 [background-5] DEBUG ch.cyberduck.core.shared.DefaultPathHomeFeature - No default path set for bookmark Host{protocol=Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}, region='null', port=80, hostname='10.11.12.31', credentials=Credentials{user='bmV3YWRtaW4=', password='', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=9223372036854775807}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=9223372036854775807}', identity=null}, uuid='7b39f090-42dd-40be-b0bd-129d1af6e7eb', nickname='null', defaultpath='null', workdir=null, custom=null, labels=null}
2026-02-16 12:35:21,513 [background-5] DEBUG ch.cyberduck.core.shared.PathAttributesHomeFeature - Read attributes for Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,513 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,517 [background-5] DEBUG ch.cyberduck.core.s3.RequestEntityRestStorageService - No default hostname set in Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}
2026-02-16 12:35:21,517 [background-5] INFO  ch.cyberduck.core.worker.MountWorker - Mount path Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,517 [background-5] INFO  ch.cyberduck.core.AbstractCache - Invalidate Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,517 [background-5] DEBUG ch.cyberduck.core.AbstractCache - No cache for Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,521 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,521 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,521 [background-5] DEBUG ch.cyberduck.core.worker.ListWorker - Run with feature VaultRegistryListService{proxy=ch.cyberduck.core.spectra.SpectraListService@14abea4}
2026-02-16 12:35:21,521 [background-5] DEBUG ch.cyberduck.core.vault.registry.VaultRegistryListService - Look for vault in Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,521 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,521 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,521 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,521 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:21,537 [background-5] INFO  com.spectralogic.ds3client.networking.NetworkClientImpl - Sending request: GET http://10.11.12.31:80 /
2026-02-16 12:35:21,557 [background-5] DEBUG com.spectralogic.ds3client.utils.Signature - String to sign: GET\n\napplication/xml\nMon, 16 Feb 2026 19:35:21 +0000\n/
2026-02-16 12:35:21,557 [background-5] DEBUG org.apache.http.client.protocol.RequestAddCookies - CookieSpec selected: default
2026-02-16 12:35:21,557 [background-5] DEBUG org.apache.http.client.protocol.RequestAuthCache - Auth cache not set in the context
2026-02-16 12:35:21,557 [background-5] WARN  ch.cyberduck.core.s3.S3AWS2SignatureRequestInterceptor - No bucket name in context org.apache.http.client.protocol.HttpClientContext@2daf2eb
2026-02-16 12:35:21,557 [background-5] DEBUG ch.cyberduck.core.s3.S3AWS2SignatureRequestInterceptor - Use bucket name  from context
2026-02-16 12:35:21,557 [background-5] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection request: [route: {}->http://10.11.12.31:80][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
2026-02-16 12:35:21,561 [background-5] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection leased: [id: 0][route: {}->http://10.11.12.31:80][total available: 0; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
2026-02-16 12:35:21,561 [background-5] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-0: set socket timeout to 0
2026-02-16 12:35:21,561 [background-5] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-0: set socket timeout to 30000
2026-02-16 12:35:21,561 [background-5] DEBUG org.apache.http.impl.execchain.MainClientExec - Executing request GET / HTTP/1.1
2026-02-16 12:35:21,561 [background-5] DEBUG org.apache.http.impl.execchain.MainClientExec - Proxy auth state: UNCHALLENGED
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - GET / HTTP/1.1
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - Host: 10.11.12.31
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - DATE: Mon, 16 Feb 2026 19:35:21 +0000
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - Content-Type: application/xml
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - User-Agent: Cyberduck/9.2.4.43667 (Windows 11/10.0.22631.0) (amd64)
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - Naming-Convention: s3
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - Authorization: ********
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - Connection: Keep-Alive
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - Accept-Encoding: gzip,deflate
2026-02-16 12:35:21,561 [background-5] INFO  ch.cyberduck.transcript.request - x-amz-date: 20260216T193521Z
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - HTTP/1.1 200 OK
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - RequestHandler-Version: 1.8D18E6B210A254D12ACD074CF265700F
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - x-amz-request-id: 3050
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - Content-Type: text/xml;charset=UTF-8
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - Content-Language: en-US
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - Content-Length: 666
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - Date: Mon, 16 Feb 2026 19:35:21 GMT
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - Keep-Alive: timeout=60
2026-02-16 12:35:21,570 [background-5] INFO  ch.cyberduck.transcript.response - Connection: keep-alive
2026-02-16 12:35:21,570 [background-5] DEBUG org.apache.http.impl.execchain.MainClientExec - Connection can be kept alive for 60000 MILLISECONDS
2026-02-16 12:35:21,573 [background-5] INFO  com.spectralogic.ds3client.networking.NetworkClientImpl - Server responded with 200 for request #3050
2026-02-16 12:35:21,573 [background-5] DEBUG com.spectralogic.ds3client.commands.parsers.utils.ResponseParserUtils - Did not find a content checksum header
2026-02-16 12:35:21,805 [background-5] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection [id: 0][route: {}->http://10.11.12.31:80] can be kept alive for 60.0 seconds
2026-02-16 12:35:21,805 [background-5] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-0: set socket timeout to 0
2026-02-16 12:35:21,805 [background-5] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection released: [id: 0][route: {}->http://10.11.12.31:80][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
2026-02-16 12:35:21,805 [background-5] INFO  ch.cyberduck.core.worker.WorkerCanceledListProgressListener - Retrieved chunk of 5 items in Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.AbstractCache - Caching Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundCallable - Return result Path{path='/', type=[directory, volume]} from background action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.threading.AbstractBackgroundAction - Finish background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundActionRegistry - Remove action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundActionRegistry - Remove action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundCallable - Invoke cleanup for background action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
2026-02-16 12:35:21,809 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundCallable - Releasing lock for background runnable WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
2026-02-16 12:35:21,809 [Thread-0] DEBUG ch.cyberduck.core.threading.WorkerBackgroundAction - Cleanup worker MountWorker{cache=Cache{size=1}}
2026-02-16 12:35:21,809 [Thread-0] WARN  ch.cyberduck.core.Collection - Skip adding duplicate Host{protocol=Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}, region='null', port=80, hostname='10.11.12.31', credentials=Credentials{user='bmV3YWRtaW4=', password='', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=9223372036854775807}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=9223372036854775807}', identity=null}, uuid='7b39f090-42dd-40be-b0bd-129d1af6e7eb', nickname='null', defaultpath='null', workdir=null, custom=null, labels=null}
2026-02-16 12:35:21,813 [Thread-0] INFO  ch.cyberduck.core.AbstractFolderHostCollection - Save bookmark Host{protocol=Profile{parent=spectra, vendor=spectralogic-http, description=Spectra S3 (HTTP), image=Local{path='C:\Users\DANIEL~1\AppData\Local\Temp\0781b4a0-8fc6-46ff-80d7-133aac40ae8a\DGjBPsiQ'}}, region='null', port=80, hostname='10.11.12.31', credentials=Credentials{user='bmV3YWRtaW4=', password='', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=9223372036854775807}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=9223372036854775807}', identity=null}, uuid='7b39f090-42dd-40be-b0bd-129d1af6e7eb', nickname='null', defaultpath='null', workdir=null, custom=null, labels=null} to Local{path='C:\Users\danielbennett\AppData\Roaming\Cyberduck\Bookmarks\7b39f090-42dd-40be-b0bd-129d1af6e7eb.duck'}
2026-02-16 12:35:21,825 [watcher-1] DEBUG ch.cyberduck.core.local.FileWatcher - Retrieved key sun.nio.fs.WindowsWatchService$WindowsWatchKey@9d8a98 from watch service ch.cyberduck.core.io.watchservice.NIOEventWatchService@1970c6a
2026-02-16 12:35:21,825 [watcher-1] INFO  ch.cyberduck.core.local.FileWatcher - Detected file system event ENTRY_MODIFY
2026-02-16 12:35:21,825 [watcher-1] INFO  ch.cyberduck.core.local.FileWatcher - Process file system event ENTRY_MODIFY for 7b39f090-42dd-40be-b0bd-129d1af6e7eb.duck
2026-02-16 12:35:21,833 [watcher-1] DEBUG ch.cyberduck.core.local.FileWatcher - Wait for key from watch service ch.cyberduck.core.io.watchservice.NIOEventWatchService@1970c6a
2026-02-16 12:35:21,878 [Thread-0] DEBUG ch.cyberduck.core.worker.Worker - Cleanup with result Path{path='/', type=[directory, volume]} and failure null
2026-02-16 12:35:21,878 [Thread-0] DEBUG ch.cyberduck.core.worker.Worker - Cleanup with result Path{path='/', type=[directory, volume]}
2026-02-16 12:35:21,894 [Thread-0] DEBUG ch.cyberduck.core.AttributedList - Sort list ch.cyberduck.core.AttributedList@47a48f14 with comparator ch.cyberduck.core.NullComparator@80c30e
2026-02-16 12:35:21,894 [Thread-0] DEBUG ch.cyberduck.core.AttributedList - Filter list ch.cyberduck.core.AttributedList@47a48f14 with filter RegexFilter{pattern=\..*}
2026-02-16 12:35:21,958 [Thread-0] DEBUG ch.cyberduck.core.AttributedList - Sort list ch.cyberduck.core.AttributedList@47a48f14 with comparator ch.cyberduck.ui.comparator.FilenameComparator@1f
2026-02-16 12:35:21,966 [Thread-0] DEBUG ch.cyberduck.core.AttributedList - Filter list ch.cyberduck.core.AttributedList@47a48f14 with filter RegexFilter{pattern=\..*}
2026-02-16 12:35:21,974 [Thread-0] DEBUG Ch.Cyberduck.Ui.Controller.BrowserController - ToggleView:File
2026-02-16 12:35:22,003 [Thread-0] DEBUG ch.cyberduck.core.AttributedList - Sort list ch.cyberduck.core.AttributedList@47a48f14 with comparator ch.cyberduck.ui.comparator.FilenameComparator@1f
2026-02-16 12:35:22,003 [Thread-0] DEBUG ch.cyberduck.core.AttributedList - Filter list ch.cyberduck.core.AttributedList@47a48f14 with filter RegexFilter{pattern=\..*}
2026-02-16 12:35:22,011 [Thread-0] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:35:22,015 [Thread-0] DEBUG ch.cyberduck.core.threading.NamedThreadFactory - Create thread for runnable java.util.concurrent.ThreadPoolExecutor$Worker@7654af[State = -1, empty queue]
2026-02-16 12:35:22,015 [Thread-0] DEBUG ch.cyberduck.core.threading.AbstractBackgroundAction - Cleanup background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=1}}}
```


--------------------------------------------------------------------------------------------------------
At 9.3.0:

```
2026-02-16 12:36:41,576 [background-5] DEBUG ch.cyberduck.core.worker.ListWorker - Run with feature VaultRegistryListService{proxy=ch.cyberduck.core.spectra.SpectraListService@17f193e}
2026-02-16 12:36:41,576 [background-5] DEBUG ch.cyberduck.core.vault.registry.VaultRegistryListService - Look for vault in Path{path='/', type=[directory, volume]}
2026-02-16 12:36:41,580 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:36:41,580 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:36:41,580 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:36:41,580 [background-5] WARN  ch.cyberduck.core.serializer.impl.dd.PlistDeserializer - Unexpected value type for serialized key Properties
2026-02-16 12:36:41,592 [background-5] INFO  com.spectralogic.ds3client.networking.NetworkClientImpl - Sending request: GET http://10.11.12.31:80 /
2026-02-16 12:36:41,616 [background-5] WARN  ch.cyberduck.core.threading.BackgroundCallable - Failure running background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
java.lang.Exception
	at ch.cyberduck.core.threading.BackgroundCallable.<init>(BackgroundCallable.java:38)
	at ch.cyberduck.core.threading.DefaultBackgroundExecutor.execute(DefaultBackgroundExecutor.java:72)
	at ch.cyberduck.core.AbstractController.background(AbstractController.java:72)
	at cli.Ch.Cyberduck.Ui.Controller.BrowserController$$$003C$$003Ec__DisplayClass233_0.<UnmountImpl>b__0(Unknown Source)
	at ch.cyberduck.core.threading.AbstractBackgroundAction.cleanup(AbstractBackgroundAction.java:112)
	at ch.cyberduck.core.threading.SessionBackgroundAction.cleanup(SessionBackgroundAction.java:135)
	at ch.cyberduck.core.threading.BackgroundCallable$1.run(BackgroundCallable.java:64)
	at cli.System.Delegate.DynamicInvokeImpl(Unknown Source)
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbackDo(Unknown Source)
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Unknown Source)
	at cli.System.Threading.ExecutionContext.RunInternal(Unknown Source)
	at cli.System.Threading.ExecutionContext.Run(Unknown Source)
	at cli.System.Threading.ExecutionContext.Run(Unknown Source)
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallback(Unknown Source)
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbacks(Unknown Source)
	at cli.System.Windows.Forms.Control.WndProc(Unknown Source)
	at cli.System.Windows.Forms.Form.WndProc(Unknown Source)
	at cli.System.Windows.Forms.NativeWindow.Callback(Unknown Source)
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source)
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source)
	at cli.System.Windows.Forms.Application$ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(Unknown Source)
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoopInner(Unknown Source)
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoop(Unknown Source)
	at cli.Ch.Cyberduck.Ui.Program.Main(Unknown Source)
Caused by: java.lang.NullPointerException
	at ch.cyberduck.core.spectra.SpectraClientBuilder$1.getCredentials(SpectraClientBuilder.java:44)
	at com.spectralogic.ds3client.networking.NetworkClientImpl$RequestExecutor.addHeaders(NetworkClientImpl.java:297)
	at com.spectralogic.ds3client.networking.NetworkClientImpl$RequestExecutor.execute(NetworkClientImpl.java:237)
	at com.spectralogic.ds3client.networking.NetworkClientImpl.getResponse(NetworkClientImpl.java:177)
	at com.spectralogic.ds3client.Ds3ClientImpl.getService(Ds3ClientImpl.java:90)
	at ch.cyberduck.core.spectra.SpectraBucketListService.list(SpectraBucketListService.java:49)
	at ch.cyberduck.core.spectra.SpectraListService.list(SpectraListService.java:41)
	at ch.cyberduck.core.vault.VaultFinderListService.list(VaultFinderListService.java:48)
	at ch.cyberduck.core.vault.registry.VaultRegistryListService.list(VaultRegistryListService.java:62)
	at ch.cyberduck.core.worker.ListWorker.run(ListWorker.java:64)
	at ch.cyberduck.core.worker.MountWorker.list(MountWorker.java:72)
	at ch.cyberduck.core.worker.MountWorker.run(MountWorker.java:60)
	at ch.cyberduck.core.worker.MountWorker.run(MountWorker.java:42)
	at ch.cyberduck.core.threading.WorkerBackgroundAction.run(WorkerBackgroundAction.java:66)
	at ch.cyberduck.core.threading.SessionBackgroundAction.run(SessionBackgroundAction.java:107)
	at ch.cyberduck.core.threading.SessionBackgroundAction$1.call(SessionBackgroundAction.java:91)
	at ch.cyberduck.core.threading.DefaultRetryCallable.call(DefaultRetryCallable.java:52)
	at ch.cyberduck.core.threading.SessionBackgroundAction.call(SessionBackgroundAction.java:85)
	at ch.cyberduck.core.threading.BackgroundCallable.run(BackgroundCallable.java:109)
	at ch.cyberduck.core.threading.BackgroundCallable.call(BackgroundCallable.java:86)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:625)
	at ch.cyberduck.core.threading.NamedThreadFactory$1.run(NamedThreadFactory.java:57)
	at java.lang.Thread.run(Thread.java:980)
2026-02-16 12:36:41,628 [background-5] WARN  ch.cyberduck.core.worker.DefaultExceptionMappingService - Map failure java.lang.NullPointerException
2026-02-16 12:36:41,632 [background-5] INFO  ch.cyberduck.core.threading.SessionBackgroundAction - Run alert callback Ch.Cyberduck.Ui.Winforms.Threading.DialogAlertCallback for failure BackgroundException{class=class ch.cyberduck.core.exception.ConnectionCanceledException, file=null, message='Error', detail='Unknown application error.', cause='java.lang.NullPointerException'}
2026-02-16 12:36:41,632 [background-5] DEBUG ch.cyberduck.core.threading.DefaultFailureDiagnostics - Determine cause for failure Error
2026-02-16 12:36:41,636 [background-5] DEBUG ch.cyberduck.core.threading.AbstractBackgroundAction - Finish background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
2026-02-16 12:36:41,636 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundActionRegistry - Remove action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
2026-02-16 12:36:41,636 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundActionRegistry - Remove action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
2026-02-16 12:36:41,636 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundCallable - Invoke cleanup for background action WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
2026-02-16 12:36:41,636 [background-5] DEBUG ch.cyberduck.core.threading.BackgroundCallable - Releasing lock for background runnable WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
2026-02-16 12:36:41,636 [Thread-0] WARN  ch.cyberduck.core.threading.WorkerBackgroundAction - Missing result for worker MountWorker{cache=Cache{size=0}}. Use default value.
2026-02-16 12:36:41,636 [Thread-0] DEBUG ch.cyberduck.core.worker.Worker - Cleanup with result null and failure BackgroundException{class=class ch.cyberduck.core.exception.ConnectionCanceledException, file=null, message='Error', detail='Unknown application error.', cause='java.lang.NullPointerException'}
2026-02-16 12:36:41,636 [Thread-0] DEBUG ch.cyberduck.core.worker.Worker - Cleanup with result null
```

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraAttributesFinderFeature.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraAttributesFinderFeature.java
@@ -63,7 +63,7 @@ public class SpectraAttributesFinderFeature implements AttributesFinder, Attribu
 
     protected HeadObjectResponse details(final Path file) throws BackgroundException {
         try {
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             final HeadObjectResponse response = client.headObject(new HeadObjectRequest(containerService.getContainer(file).getName(), containerService.getKey(file)));
             switch(response.getStatus()) {
                 case DOESNTEXIST:

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraBucketListService.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraBucketListService.java
@@ -44,7 +44,7 @@ public class SpectraBucketListService implements ListService {
     public AttributedList<Path> list(final Path directory, final ListProgressListener listener) throws BackgroundException {
         try {
             final AttributedList<Path> buckets = new AttributedList<Path>();
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             final GetServiceResponse response = client.getService(new GetServiceRequest());
             for(final BucketDetails b : response.getListAllMyBucketsResult().getBuckets()) {
                 final Path bucket = new Path(PathNormalizer.normalize(b.getName()), EnumSet.of(Path.Type.volume, Path.Type.directory));

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraBulkService.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraBulkService.java
@@ -101,7 +101,7 @@ public class SpectraBulkService implements Bulk<Set<UUID>> {
      */
     @Override
     public Set<UUID> pre(final Transfer.Type type, final Map<TransferItem, TransferStatus> files, final ConnectionCallback callback) throws BackgroundException {
-        final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+        final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
         final Map<Path, List<Ds3Object>> objects = new HashMap<Path, List<Ds3Object>>();
         for(Map.Entry<TransferItem, TransferStatus> item : files.entrySet()) {
             final Path file = item.getKey().remote;
@@ -193,7 +193,7 @@ public class SpectraBulkService implements Bulk<Set<UUID>> {
             }
             final String job = status.getParameters().get(REQUEST_PARAMETER_JOBID_IDENTIFIER);
             log.debug("Cancel job {}", job);
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             client.cancelJobSpectraS3(new CancelJobSpectraS3Request(job));
         }
         catch(FailedRequestException e) {
@@ -227,7 +227,7 @@ public class SpectraBulkService implements Bulk<Set<UUID>> {
             final String job = status.getParameters().get(REQUEST_PARAMETER_JOBID_IDENTIFIER);
             log.debug("Query status for job {}", job);
             // Fetch current list from server
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             // For GET, the client may need to issue multiple GET requests for a single object if it has
             // been broken up into multiple pieces due to its large size
             // For PUT, This will allocate a working window of job chunks, if possible, and return a list of
@@ -329,7 +329,7 @@ public class SpectraBulkService implements Bulk<Set<UUID>> {
     protected void clear() throws BackgroundException {
         try {
             // Cancel all active jobs to remove references to cached objects
-            final Ds3Client ds3Client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client ds3Client = new SpectraClientBuilder().wrap(session, session.getHost());
             ds3Client.cancelAllActiveJobsSpectraS3(new CancelAllActiveJobsSpectraS3Request());
             // Clear cache
             final RequestEntityRestStorageService client = session.getClient();

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraClientBuilder.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraClientBuilder.java
@@ -21,7 +21,6 @@ import ch.cyberduck.core.Scheme;
 import ch.cyberduck.core.preferences.HostPreferencesFactory;
 
 import org.apache.http.impl.client.CloseableHttpClient;
-import org.jets3t.service.impl.rest.httpclient.RestStorageService;
 
 import java.net.URI;
 
@@ -32,7 +31,7 @@ import com.spectralogic.ds3client.networking.ConnectionDetails;
 import com.spectralogic.ds3client.networking.NetworkClientImpl;
 
 public class SpectraClientBuilder {
-    public Ds3Client wrap(final RestStorageService client, final Host bookmark) {
+    public Ds3Client wrap(final SpectraSession session, final Host bookmark) {
         return new Ds3ClientImpl(new NetworkClientImpl(new ConnectionDetails() {
             @Override
             public String getEndpoint() {
@@ -41,8 +40,8 @@ public class SpectraClientBuilder {
 
             @Override
             public Credentials getCredentials() {
-                return new Credentials(client.getProviderCredentials().getAccessKey(),
-                    client.getProviderCredentials().getSecretKey());
+                return new Credentials(session.getAuthentication().get().getTokens().getAccessKeyId(),
+                        session.getAuthentication().get().getTokens().getSecretAccessKey());
             }
 
             @Override
@@ -84,6 +83,6 @@ public class SpectraClientBuilder {
             public String getUserAgent() {
                 return new PreferencesUseragentProvider().get();
             }
-        }, (CloseableHttpClient) client.getHttpClient()));
+        }, (CloseableHttpClient) session.getClient().getHttpClient()));
     }
 }

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraDeleteFeature.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraDeleteFeature.java
@@ -54,7 +54,7 @@ public class SpectraDeleteFeature implements Delete {
     @Override
     public void delete(final Map<Path, TransferStatus> files, final PasswordCallback prompt, final Callback callback) throws BackgroundException {
         try {
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             final Map<Path, TransferStatus> filtered = new LinkedHashMap<>(files);
             for(Iterator<Map.Entry<Path, TransferStatus>> iter = filtered.entrySet().iterator(); iter.hasNext(); ) {
                 final Map.Entry<Path, TransferStatus> file = iter.next();

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraFindFeature.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraFindFeature.java
@@ -48,7 +48,7 @@ public class SpectraFindFeature implements Find {
             return true;
         }
         try {
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             if(containerService.isContainer(file)) {
                 final HeadBucketResponse response = client.headBucket(new HeadBucketRequest(containerService.getContainer(file).getName()));
                 switch(response.getStatus()) {

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraObjectListService.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraObjectListService.java
@@ -69,7 +69,7 @@ public class SpectraObjectListService extends S3AbstractListService {
         try {
             final String prefix = this.createPrefix(directory);
             final AttributedList<Path> objects = new AttributedList<>();
-            final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+            final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
             final Path bucket = containerService.getContainer(directory);
             long revision = 0L;
             String marker = null;

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraSession.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraSession.java
@@ -14,17 +14,20 @@
 
 package ch.cyberduck.core.spectra;
 
+import ch.cyberduck.core.Credentials;
 import ch.cyberduck.core.DisabledUrlProvider;
 import ch.cyberduck.core.Host;
 import ch.cyberduck.core.HostKeyCallback;
 import ch.cyberduck.core.ListService;
 import ch.cyberduck.core.LoginCallback;
+import ch.cyberduck.core.TemporaryAccessTokens;
 import ch.cyberduck.core.UrlProvider;
 import ch.cyberduck.core.cdn.DistributionConfiguration;
 import ch.cyberduck.core.exception.BackgroundException;
 import ch.cyberduck.core.features.*;
 import ch.cyberduck.core.proxy.ProxyFinder;
 import ch.cyberduck.core.s3.RequestEntityRestStorageService;
+import ch.cyberduck.core.s3.S3CredentialsStrategy;
 import ch.cyberduck.core.s3.S3Session;
 import ch.cyberduck.core.shared.DefaultDownloadFeature;
 import ch.cyberduck.core.shared.DisabledMoveFeature;
@@ -32,6 +35,7 @@ import ch.cyberduck.core.ssl.X509KeyManager;
 import ch.cyberduck.core.ssl.X509TrustManager;
 import ch.cyberduck.core.threading.CancelCallback;
 
+import org.apache.http.impl.client.HttpClientBuilder;
 import org.jets3t.service.Jets3tProperties;
 
 public class SpectraSession extends S3Session {
@@ -46,6 +50,29 @@ public class SpectraSession extends S3Session {
         final Jets3tProperties configuration = client.getConfiguration();
         configuration.setProperty("s3service.enable-storage-classes", String.valueOf(false));
         return client;
+    }
+
+    public SpectraCredentialsStrategy getAuthentication() {
+        return (SpectraCredentialsStrategy) super.getAuthentication();
+    }
+
+    @Override
+    protected SpectraCredentialsStrategy configureCredentialsStrategy(final HttpClientBuilder configuration, final LoginCallback prompt) {
+        return new SpectraCredentialsStrategy(host.getCredentials());
+    }
+
+    protected final class SpectraCredentialsStrategy implements S3CredentialsStrategy {
+        final Credentials credentials;
+
+        public SpectraCredentialsStrategy(final Credentials credentials) {
+            this.credentials = credentials
+                    .setTokens(new TemporaryAccessTokens(host.getCredentials().getUsername(), host.getCredentials().getPassword()));
+        }
+
+        @Override
+        public Credentials get() {
+            return credentials;
+        }
     }
 
     @Override

--- a/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraVersioningFeature.java
+++ b/spectra/src/main/java/ch/cyberduck/core/spectra/SpectraVersioningFeature.java
@@ -62,7 +62,7 @@ public class SpectraVersioningFeature implements Versioning {
 
     @Override
     public VersioningConfiguration getConfiguration(final Path file) throws BackgroundException {
-        final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+        final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
         final Path container = containerService.getContainer(file);
         if(container.isRoot()) {
             return VersioningConfiguration.empty();
@@ -98,7 +98,7 @@ public class SpectraVersioningFeature implements Versioning {
                 }
                 else {
                     log.debug("Enable bucket versioning for {}", container);
-                    final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+                    final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
                     final GetBucketSpectraS3Response bucket = client.getBucketSpectraS3(new GetBucketSpectraS3Request(container.getName()));
                     final UUID id = bucket.getBucketResult().getDataPolicyId();
                     client.modifyDataPolicySpectraS3(new ModifyDataPolicySpectraS3Request(id).withVersioning(VersioningLevel.KEEP_MULTIPLE_VERSIONS));
@@ -127,7 +127,7 @@ public class SpectraVersioningFeature implements Versioning {
 
     @Override
     public void revert(final Path file) throws BackgroundException {
-        final Ds3Client client = new SpectraClientBuilder().wrap(session.getClient(), session.getHost());
+        final Ds3Client client = new SpectraClientBuilder().wrap(session, session.getHost());
         final Path container = containerService.getContainer(file);
         try {
             client.undeleteObjectSpectraS3(new UndeleteObjectSpectraS3Request(container.getName(), containerService.getKey(file)));

